### PR TITLE
Switch CI to GitHub actions

### PR DESCRIPTION
**Pull request by omus**

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+      - uses: julia-actions/julia-buildpkg@latest
+      - run: |
+          julia --project=docs -e '
+            using Pkg
+            Pkg.develop(PackageSpec(path=pwd()))
+            Pkg.instantiate()'
+      - run: julia --project=docs docs/deploy.jl
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          DOCUMENTER_KEY: ${{ secrets.DOCUMENTER_KEY }}
```
Comment (mattBrzezinski) [inline .github/workflows/CI.yml:135]:
```suggestion
          DOCUMENTER_KEY: ${{ secrets.DOCUMENTER_KEY }}

```

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        arch:
+          - x64
+          - x86
+        exclude:
+          # Test 32-bit only on Linux
+          - os: macOS-latest
+            arch: x86
+          - os: windows-latest
+            arch: x86
+    steps:
```
Comment (mattBrzezinski) [inline .github/workflows/CI.yml:35]: Do we want Slack notifications for this project in #nightly-dev?

**Review by mattBrzezinski — APPROVED** [review]

File: .github/workflows/CI.yml (hunk 2 above)
Comment (omus) [reply .github/workflows/CI.yml:35]: We could. Typically I just handle failures on this repo which I'd notice by using CCMenu. Looks like I'll need to find an alternative.

Comment (mattBrzezinski) [reply .github/workflows/CI.yml:35]: Oo CCMenu is neat! Yeah not too sure if you want to open that up since you're the defacto maintainer of this package.

Comment (omus) [reply .github/workflows/CI.yml:35]: I'm not sure what I want to do at the moment. I'm going to punt on this as I don't want to hold up: https://github.com/JuliaTime/TimeZones.jl/pull/315

Comment (mattBrzezinski) [reply .github/workflows/CI.yml:35]: Fair enough, can always come back to it in the future!

**Review by mattBrzezinski — APPROVED** [review]

**Pull request merged**